packages/interface-ipfs-core/type/pin/remote/service.ts

```diff
@@ -0,0 +1,74 @@
+import { AbortOptions } from '../../basic'
+
+export interface API {
+  /**
+   * Registers remote pinning service with a given name. Errors if service
+   * with the given name is already registered.
+   */
+  add(name: string, credentials:Credentials & AbortOptions): Promise<void>
+
+  /**
+   * Unregisteres remote pinning service with a given name. If service with such
+   * name isn't registerede this is a noop.
+   */
+  rm(name: string, options?:AbortOptions):Promise<void>
+
+  /**
+   * List registered remote pinning services.
+   */
+  ls(options:ListOptions & AbortOptions):Promise<RemotePinService[]>
+}
+
+export interface Credentials {
+  /**
+   * Service URL
+   */
+  url: URL
```

Comment:
Can we rename `url` to `endpoint` to follow [upstream](https://github.com/ipfs/go-ipfs-config/pull/113/files#diff-177fdc882de9eb4633aafaa24834e81a1d30fad53933a66d567a1512ea00014dR18
) fields names?

```json
  "Pinning": {

    "RemoteServices": {
      "test-service": {
        "Api": {
          "Endpoint": "https://testapi.example.com/",
          "Key": "secret"
        }
      },
```


(In the future we may want to add support for libp2p and multiaddrs, and `url` will be awkward)